cluster/ci/origin/logging-infra/entrypoint.go

```diff
@@ -0,0 +1,135 @@
+// entrypoint acts as an entrypoint for containers from which
+// we wish to stream logs to GCS and whose exit should trigger
+// other actions like artifact upload
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"strconv"
+	"sync"
+	"syscall"
+)
+
+const INTERNAL_ERROR = 127
+
+func main() {
+	if len(os.Args) < 3 {
+		log.Fatalf("USAGE: %s CONFIGURATION EXECUTABLE [ARGS...]", os.Args[0])
+	}
+
+	config, err := loadConfig(os.Args[1])
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	output, group, err := setupOutputStreaming(config.ProcessLog)
```

Comment:
You can avoid some complexity (with the pipe and all) by just using a MultiWriter that writes to both stdout and the log file.